Add more documentation about the spatial track matching plugin

maarzt opened the pull request:
It was long overdue by me to describe the algorithm more in detail.

maarzt committed:
Improve documentation for spatial track matching

maarzt committed:
Changes to the spatial track matching documentation

stefanhahmann commented on docs/partC/tomancak/spatialtrackmatching.md:
```diff
@@ -1,18 +1,153 @@
+
+The plugin is included in the **"Mastodon-Tomancak"** collection of plugins for Mastodon.
+To install it in Fiji, activate the update sites: **Mastodon** and **Mastodon-Tomancak**.\
+(For help read the guide on [how to follow an update site](https://imagej.net/update-sites/following).)
```
This part is not required. There are already installation instructions for the whole Mastodon-Tomancak plugin here: https://mastodon.readthedocs.io/en/latest/docs/partC/tomancak/installation.html

This description is a sub section of Tomancak in the docs. Thus this part, is a repition.

maarzt replied:
I removed this section as suggested.

stefanhahmann commented on docs/partC/tomancak/spatialtrackmatching.md:
```diff
@@ -1,18 +1,153 @@
+
+## Usage
+
+First make sure to have the update sites installed and restart Fiji.
```
I would delete this paragraph, it feels not really required to me.

maarzt replied:
I removed this sentence as suggested.

stefanhahmann commented on docs/partC/tomancak/spatialtrackmatching.md:
```diff
@@ -1,18 +1,153 @@
+1. Select your projects as project A and project B.
+1. (Optional: If the datasets start at different stages specify a "first frame" to align them. Any data before this frame is ignored.)
+
+Now by click one of the menu availabe buttons, on of these actions can be performed:
```
```suggestion
Now by clicking one of the available buttons, these actions can be performed:
```

stefanhahmann commented on docs/partC/tomancak/spatialtrackmatching.md:
```diff
@@ -1,18 +1,153 @@
+1. In the Mastodon menu navigate to `Plugins > Lineage Analysis > Spatial Track Matching`.\
+   A dialog box, "Spatial Track Matching Across Two Mastodon Projects," will appear. 
+1. Select your projects as project A and project B.
+1. (Optional: If the datasets start at different stages specify a "first frame" to align them. Any data before this frame is ignored.)
```
```suggestion
1. (Optional: If the datasets start at different stages, specify a "first frame" to align them. Any data before this frame is ignored.)
```

stefanhahmann commented on docs/partC/tomancak/spatialtrackmatching.md:
```diff
@@ -1,18 +1,153 @@
+To use the plugin:
+1. Open two Mastodon datasets containing stereotypically developing embryos.\
+   Example datasets can be found [here](https://github.com/mastodon-sc/mastodon-example-data/blob/master/astec/).
+1. Make sure to have this initial cells named with corresponding names in both dataset.\
```
```suggestion
1. Make sure to have the initial cells named with corresponding names in both dataset.\
```

stefanhahmann commented on docs/partC/tomancak/spatialtrackmatching.md:
```diff
@@ -1,18 +1,153 @@
+---
+Here’s the revised and polished version of your text. I've improved grammar, flow, and clarity while maintaining the technical details:
+
+---
```
This seems to be an artefact from some correspondence with a chat bot. I think, this should be deleted.

maarzt replied:
Thank you for spotting this!

stefanhahmann commented on docs/partC/tomancak/spatialtrackmatching.md:
```diff
@@ -1,18 +1,153 @@
+1. X1 corresponds to Y1, and X2 corresponds to Y2.
+2. X1 corresponds to Y2, and X2 corresponds to Y1.
+
+To determine the correct correspondence, our strategy is to align the embryos with one another.
```
```suggestion
To determine the correct correspondence, our strategy is to spatially align the embryos with one another.
```

I think, this would be even clearer

stefanhahmann commented on docs/partC/tomancak/spatialtrackmatching.md:
```diff
@@ -1,18 +1,153 @@
+
+This simple yet effective approach lets us find the correspondence for a cell division, provided we already know the correspondence for the parent cells. By repeating this process for all cell divisions in the Mastodon datasets, we can establish the full set of correspondences between the two datasets.
+
+Certainly! Here’s an improved version of your paragraph with clearer phrasing, improved grammar, and enhanced readability:
```
I think, this should be deleted as well. Reads again like an artefact of conversation with a chat bot.

maarzt replied:
Thank you for spotting this!

stefanhahmann commented on docs/partC/tomancak/spatialtrackmatching.md:
```diff
@@ -1,18 +1,153 @@
+
+There is no guarantee that the determined correspondences are biologically accurate or relevant. However, the method provides an indication of the quality of the computed correspondences. By analyzing the angle between the cell division directions, we can assess their reliability. Angles close to 0° or 180° strongly suggest a specific correspondence, whereas angles near 90° offer only a weak dist ...
+
+The Spatial Track Matching plugin includes an option to plot cell division angles. Below is an example of such a plot:
```
```suggestion
The Spatial Track Matching dialog includes an option to plot cell division angles (<i>"plot angles"</i>). Below is an example of such a plot:
```

stefanhahmann commented on docs/partC/tomancak/spatialtrackmatching.md:
```diff
@@ -1,18 +1,153 @@
+
+This approach is more dynamic and works even if the embryos rotate during the recording. It starts with the correspondence between the initial root cells (e.g., A ↔ A, B ↔ B, C ↔ C, D ↔ D). 
+
+For each root cell (e.g., cell A), instead of extracting a fixed position vector. We extrac a position vector by timepoint which will be denoted as P(A,X,t), where A is the cell, X is the embrbryo and t is the timepoint. If the timepoint t is between the appearance of the cell A and there cell division P(A,X,t) will simple denote the position vector of the cell A. Afterwards when the cell divided ...
```
```suggestion
For each root cell (e.g., cell A), instead of extracting a fixed position vector. We extract a position vector by timepoint which will be denoted as P(A,X,t), where A is the cell, X is the embryo and t is the timepoint. If the timepoint t is between the appearance of the cell A and its cell division, P(A,X,t) will simply denote the position vector of the cell A. Afterwards when the cell has divided into two daugther cells, it will be the average position of the two daugther cells of cell A at that time point. When the daughter cells divide further, it will simply be the average position of all the descendants of cell A at the time point in embryo A.
```

stefanhahmann commented on docs/partC/tomancak/spatialtrackmatching.md:
```diff
@@ -1,18 +1,153 @@
+
+### Remarks
+
+In the first timepoints after cell division the daugther cells usually move a lot. To reduce the noise when compution a cell division direction, we avarage over the position of the daugther cell in the three time points after cell division. This has the effect the the "timepoint of the cell division direction" is t + 2. If t denotes last timepoint before cell division. This is also the reason why ...
```
```suggestion
In the first timepoints after cell division the daugther cells usually move a lot. To reduce the noise when computing a cell division direction, we avarage over the position of the daugther cell in the three time points after cell division. This has the effect that the "timepoint of the cell division direction" is t + 2, if t denotes the last timepoint before cell division. This is also the reason why an offset t+2 is used when computing the dynamic spatial registration.
```

stefanhahmann reviewed (changes requested):
This PR improves the explanation for the spatial track matching a lot.

I have some suggestions to the PR:
* Fix some typos as indicated in the review.
* Remove the installation instructions (they would be duplicated, as they exist already here: https://mastodon.readthedocs.io/en/latest/docs/partC/tomancak/installation.html
* Remove the artefacts of conversations with chat bots. ;)

maarzt committed:
Apply suggestions from code review

Co-authored-by: Stefan Hahmann <stefanhahmann@users.noreply.github.com>

maarzt committed:
Remove accidentally added comments from spatial track matching documentation

maarzt committed:
Remove installation remarks from spatial track matching documentation

This change was suggested in the PR review. The installation instructions
for mastodon tomancak plugins are in another centralized place.

maarzt committed:
Spatial track matching documentation: reformulate sentence

maarzt commented:
Hello @stefanhahmann,

Thank you for reviewing this PR so carefully. I applied all changes as they were suggested. And changed a few more words in the last commit 6888cb50d1975fb05c37b30726e6a49fa38ae6e6.

Do you think this is ready to be merged now?

maarzt committed:
Spatial track matching: fix typos and reformulations

maarzt committed:
Add remark on term "root cell"

stefanhahmann reviewed (approved):
I think this can be merged now.

stefanhahmann commented:
> Hello @stefanhahmann,
> 
> Thank you for reviewing this PR so carefully. I applied all changes as they were suggested. And changed a few more words in the last commit [6888cb5](https://github.com/mastodon-sc/mastodon-documentation/commit/6888cb50d1975fb05c37b30726e6a49fa38ae6e6).
> 
> Do you think this is ready to be merged now?

Yes
